### ISSUE TITLE
Pulling the latest release (Docs outdated?)

### ISSUE THREAD

**Issue by jparnell8839** [octokit/octokit.net#1531]
I have been trying to get some semblance of the releases down, and it just doesn't seem to be working at all. I'm pretty sure that as Octokit has been updated, the docs have fallen asunder.

For instance, the following code, from the docs:

```
static void Main(string[] args)
{
	var client = new GitHubClient(new ProductHeaderValue("my-cool-app"));

	var releases = client.Release.GetAll("octokit", "octokit.net");
	var latest = releases[0];
	Console.WriteLine(
		"The latest release is tagged at {0} and is named {1}",
		latest.TagName,
		latest.Name);

	Console.ReadKey();
}
```

results in the following errors:
```
Error	CS1061	'GitHubClient' does not contain a definition for 'Release' and no extension method 'Release' accepting a first argument of type 'GitHubClient' could be found (are you missing a using directive or an assembly reference?)
Error	CS0021	Cannot apply indexing with [] to an expression of type 'Task<IReadOnlyList<T>>'
```
I am using strictly what is in the documentation, i.e. the `var client = new GitHubClient(new ProductHeaderValue("my-cool-app"));` from [Getting Started](http://octokitnet.readthedocs.io/en/latest/getting-started/) and the rest from [Releases](http://octokitnet.readthedocs.io/en/latest/releases/#get-all)

So, how would I be able to gather info about the latest release from a given repo with the intent to download an asset from that release?

**Comment by shiftkey** [comment]
@jparnell8839 ah yes, I know what's happened here. Here's the commit where we removed the code, which previously had an Obsolete message: https://github.com/octokit/octokit.net/commit/2cc2ccda622e0774094e8449802eb63c2044da98

Originally we had `client.Release` but we wanted to align the Octokit API with how the GitHub documentation organizes it's endpoints. This meant moving it under `Repository`, like this page: https://developer.github.com/v3/repos/releases/

Of course, this doesn't address how we keep the documentation in sync for when the API changes. I don't have any good answers on this - but we can currently test Linqpad snippets against the real binaries and catch failures at the CI stage, if that's something we want to leverage more broadly...

**Comment by jparnell8839** [comment]
How would I call that? I'm not familiar with the GitHub API and/or how to read it. Am I anywhere close to the nail?
```
static void Main(string[] args)
{
	var client = new GitHubClient(new ProductHeaderValue("my-cool-app"));
	var releases = client.Repository.Release.GetAll("octokit","ocktokit.net");
	Console.ReadKey();
}
```

**Comment by ryangribble** [comment]
Yep that's it.  Basically it was just relocated from `client.Release` to `client.Repository.Release`

It's an interesting point though about how to keep documented code samples up to date.  I guess some sort of system where the linqpad snippets (which we can automatically test) are embedded into documentation... But that implies a built docs system whereas currently we really just have a limited number of markdown files

**Comment by jparnell8839** [comment]
Well, that was a fun bit of troubleshooting, but I got it to work... it needs await, so I had to put it in an async method. And you cannot make Main() asynchronous.

So for any Google stumblers, here's the code:

```
using System;
using Octokit;

namespace ConsoleApplication1
{
    class Program
    {
        static void Main(string[] args)
        {
            GetReleases();
            Console.ReadKey();
        }

        static async void GetReleases()
        {
            try
            {
                var client = new GitHubClient(new ProductHeaderValue("my-cool-app"));
                var releases = await client.Repository.Release.GetAll("octokit", "octokit.net");
                var latest = releases[0];
                Console.WriteLine(
                           "The latest release is tagged at {0} and is named {1}",
                           latest.TagName,
                           latest.Name);
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
            }

        }
    }
}
```

**Comment by ryangribble** [comment]
Glad you got it working @jparnell8839 

Just to reply to this:
>  it needs await, so I had to put it in an async method. And you cannot make Main() asynchronous.

FYI in c# you can force an async call to be run in a blocking fashion (and thus can be run from your Main() method, by using the `.Result` like this:
`var releases = client.Repository.Release.GetAll("octokit", "octokit.net").Result;`


Also in terms of this:
>(because "ocktokit" doesn't have any releases, go figure)

This doesn't sound right as there are a number of octokit releases!  When I run some sample code based off what you posted, I get the expected 48 releases
![image](https://cloud.githubusercontent.com/assets/5425163/21849483/c4b1cc46-d851-11e6-9476-7013c4219725.png)

**Comment by jparnell8839** [comment]
@ryangribble  I did not know about the .Result call... good info to have! Asynchronous methods are something new to me.

And yes, you're right, "octokit" has many releases. "oc**k**tokit", on the other hand does not. Typos will getcha every time, and I'm glad I got one in my first testing, because while this was a simple typo, there may actually be a project that doesn't have any releases, and will throw an exception.
